### PR TITLE
ARROW-3440: [Gandiva] fix readme for builds

### PR DESCRIPTION
**Pull request by pravindra**
- also use thirdy-party protobuf install for gandiva

### CODE REVIEW

**Review by wesm — COMMENTED** [review]
Left some comments. Another way to trigger builds without making trivial changes is to `git commit --amend` to change the git hash and then `git push -f`

File: cpp/README.md [hunk 1]
```diff
@@ -225,6 +225,37 @@ The optional arrow reader for the Apache ORC format (found in the
 This is currently not supported on windows. Note that this functionality is
 still in Alpha stages, and subject to API changes without deprecation warnings.
 
+### Building and developing Gandiva (optional)
+
+In addition to the arrow dependancies, gandiva requires :
```
[the remark this answers is not in the document]
Comment (pravindra) [reply cpp/README.md]: will fix

File: cpp/README.md [hunk 2]
```diff
@@ -225,6 +225,37 @@ The optional arrow reader for the Apache ORC format (found in the
...
+* On linux, gcc 4.9 or higher C++11-enabled compiler.
```
[the remark this answers is not in the document]
Comment (pravindra) [reply cpp/README.md]: currently, yes. We are getting linker errors with gcc 4.8 - I'll create a jira to resolve that.

File: cpp/README.md [hunk 3]
```diff
@@ -225,6 +225,37 @@ The optional arrow reader for the Apache ORC format (found in the
 This is currently not supported on windows. Note that this functionality is
 still in Alpha stages, and subject to API changes without deprecation warnings.
 
+### Building and developing Gandiva (optional)
```
[the remark this answers is not in the document]
Comment (pravindra) [reply cpp/README.md]: will add a one-line description.

File: cpp/src/gandiva/jni/CMakeLists.txt [hunk 4]
```diff
@@ -17,9 +17,7 @@
 
 project(gandiva_jni)
 
-# Find protobuf
 set(Protobuf_USE_STATIC_LIBS "ON")
```
[the remark this answers is not in the document]
Comment (pravindra) [reply cpp/src/gandiva/jni/CMakeLists.txt]: will remove

File: cpp/README.md [hunk 5]
```diff
@@ -225,6 +225,37 @@ The optional arrow reader for the Apache ORC format (found in the
...
+```
+
+The optional `gandiva` libraries and tests can be built by passing
+`-DARROW_GANDIVA=on`.
+
+```shell
+cmake .. -DARROW_GANDIVA=on
+make
+ctest -L unitest
+ctest -L integ
```
[the remark this answers is not in the document]
Comment (pravindra) [reply cpp/README.md]: will do

File: cpp/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (pravindra) [reply cpp/README.md]: created ARROW-3501.

File: java/README.md [hunk 6]
```diff
@@ -25,13 +25,25 @@ install:
...
 mvn install
 ```
 
+## Building and running tests for gandiva (optional)
+
+[Gandiva cpp][2] must be built before this step.
+
+```
+cd java
+mvn install -P gandiva -pl gandiva -am
```
Comment (praveenbingo) [inline java/README.md]: please add the gandiva cpp directory too..

Comment (pravindra) [reply java/README.md]: done

**Comment by pravindra** [comment]
@wesm - I've addressed all the review comments and the build passed. Can you please merge this change ?

**Review by wesm — APPROVED** [review]
+1, thanks!

**Pull request closed without merging**